RF: dandi/cli to avoid circular import

probably to move all helper functions in a dedicated `cli/base.py` etc.

Otherwise this demonstrates them:
```
$> python -m pytest -v --doctest-modules dandi/cli
======================================================================================================= test session starts =======================================================================================================
platform linux -- Python 3.8.5, pytest-4.6.9, py-1.8.1, pluggy-0.13.0 -- /home/yoh/proj/dandi/dandi-cli/venvs/dev3/bin/python
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/home/yoh/proj/dandi/dandi-cli/.hypothesis/examples')
rootdir: /home/yoh/proj/dandi/dandi-cli, inifile: pytest.ini
plugins: mock-3.2.0, cov-2.10.0, xdist-1.32.0, timeout-1.3.3, forked-1.1.3, hypothesis-4.36.2
collected 8 items / 2 errors / 6 selected                                                                                                                                                                                         

============================================================================================================= ERRORS ==============================================================================================================
___________________________________________________________________________________________ ERROR collecting dandi/cli/cmd_download.py ____________________________________________________________________________________________
dandi/cli/cmd_download.py:4: in <module>
    from .command import devel_option, main, map_to_click_exceptions
dandi/cli/command.py:222: in <module>
    from .cmd_download import download  # noqa: E402, F401
E   ImportError: cannot import name 'download' from partially initialized module 'dandi.cli.cmd_download' (most likely due to a circular import) (/home/yoh/proj/dandi/dandi-cli/dandi/cli/cmd_download.py)
___________________________________________________________________________________________ ERROR collecting dandi/cli/cmd_register.py ____________________________________________________________________________________________
dandi/cli/cmd_register.py:3: in <module>
    from .command import (
dandi/cli/command.py:224: in <module>
    from .cmd_register import register  # noqa: E402, F401
E   ImportError: cannot import name 'register' from partially initialized module 'dandi.cli.cmd_register' (most likely due to a circular import) (/home/yoh/proj/dandi/dandi-cli/dandi/cli/cmd_register.py)
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
===================================================================================================== 2 error in 0.16 seconds =============================================================
```

--- a/dandi/cli/base.py
+++ b/dandi/cli/base.py
@@ -1,0 +1,129 @@
+from functools import wraps
+import os
+from os import path as op
+import click
+from .. import get_logger, set_logger_level  # noqa: F401
+from ..consts import dandiset_metadata_file, known_instances
+
+lgr = get_logger()
+
+# Aux common functionality
+
+
+def get_files(paths, recursive=True, recursion_limit=None):
+    """Given a list of paths, return a list of paths
+    """
+    # For now we support only individual files
+    dirs = list(filter(op.isdir, paths))
+    if dirs:
+        raise NotImplementedError(
+            "ATM supporting only listing of individual files, no recursive "
+            "operation. Was provided following directories: {}".format(", ".join(dirs))
+        )
+    return paths
+
+
+# ???: could make them always available but hidden
+#  via  hidden=True.
+def devel_option(*args, **kwargs):
+    """A helper to make command line options useful for development (only)
+
+    They will become available..."""
+
+    def wrapper(f):
+        if not os.environ.get("DANDI_DEVEL", None):
+            return f
+        else:
+            return click.option(*args, **kwargs)(f)
+
+    return wrapper
+
+
+#
+# Common options to reuse
+#
+# Functions to provide customizations where needed
+def _updated_option(*args, **kwargs):
+    args, d = args[:-1], args[-1]
+    kwargs.update(d)
+    return click.option(*args, **kwargs)
+
+
+def dandiset_id_option(**kwargs):
+    return _updated_option(
+        "--dandiset-id",
+        kwargs,
+        help=f"ID of the dandiset on DANDI archive.  Necessary to populate "
+        f"{dandiset_metadata_file}. Please use 'register' command to first "
+        f"register a new dandiset.",
+        # TODO: could add a check for correct regexp
+    )
+
+
+def dandiset_path_option(**kwargs):
+    return _updated_option(
+        "-d",
+        "--dandiset-path",
+        kwargs,
+        help="Top directory (local) of the dandiset.",
+        type=click.Path(exists=True, dir_okay=True, file_okay=False),
+    )
+
+
+def instance_option():
+    return devel_option(
+        "-i",
+        "--dandi-instance",
+        help="For development: DANDI instance to use",
+        type=click.Choice(sorted(known_instances)),
+        default="dandi",
+        show_default=True,
+    )
+
+
+def devel_debug_option():
+    return devel_option(
+        "--devel-debug",
+        help="For development: do not use pyout callbacks, do not swallow"
+        " exception, do not parallize",
+        default=False,
+        is_flag=True,
+    )
+
+
+def map_to_click_exceptions(f):
+    """Catch all exceptions and re-raise as click exceptions.
+
+    Will be active only if DANDI_DEVEL is not set and --pdb is not given
+    """
+
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        import girder_client as gcl
+        from ..girder import get_HttpError_response
+
+        try:
+            return f(*args, **kwargs)
+        # Prints global Usage: useless in majority of cases.
+        # It seems we better use it with some ctx, so it would hint in some
+        # cases to the help of a specific command
+        # except ValueError as e:
+        #     raise click.UsageError(str(e))
+        except Exception as e:
+            if isinstance(e, gcl.HttpError):
+                resp = get_HttpError_response(e)
+                if resp is None:
+                    e_str = str(e)
+                else:
+                    e_str = resp.get("message", str(e))
+            else:
+                e_str = str(e)
+            lgr.debug("Caught exception %s", e_str)
+            if not map_to_click_exceptions._do_map:
+                raise
+            raise click.ClickException(e_str)
+
+    return wrapper
+
+
+map_to_click_exceptions._do_map = not bool(os.environ.get("DANDI_DEVEL", None))

--- a/dandi/cli/cmd_download.py
+++ b/dandi/cli/cmd_download.py
@@ -1,10 +1,10 @@
 import os
 
 import click
-from .command import devel_option, main, map_to_click_exceptions
+from .base import devel_option, map_to_click_exceptions
 
 
-@main.command()
+@click.command()
 @click.option(
     "-o",
     "--output-dir",

--- a/dandi/cli/cmd_ls.py
+++ b/dandi/cli/cmd_ls.py
@@ -4,14 +4,14 @@ import time
 
 import click
 
-from .command import lgr, main, map_to_click_exceptions
+from .base import lgr, map_to_click_exceptions
 
 from ..utils import safe_call
 
 # TODO: all the recursion options etc
 
 
-@main.command()
+@click.command()
 @click.option(
     "-F",
     "--fields",

--- a/dandi/cli/cmd_organize.py
+++ b/dandi/cli/cmd_organize.py
@@ -2,17 +2,11 @@ import click
 import os
 import os.path as op
 
-from .command import (
-    dandiset_path_option,
-    devel_debug_option,
-    lgr,
-    main,
-    map_to_click_exceptions,
-)
+from .base import dandiset_path_option, devel_debug_option, lgr, map_to_click_exceptions
 from ..consts import dandiset_metadata_file, file_operation_modes
 
 
-@main.command()
+@click.command()
 @dandiset_path_option(
     help="A top directory (local) of the dandiset to organize files under. "
     "If not specified, dandiset current directory is under is assumed. "

--- a/dandi/cli/cmd_register.py
+++ b/dandi/cli/cmd_register.py
@@ -1,14 +1,9 @@
 import click
 
-from .command import (
-    dandiset_path_option,
-    instance_option,
-    main,
-    map_to_click_exceptions,
-)
+from .base import dandiset_path_option, instance_option, map_to_click_exceptions
 
 
-@main.command()
+@click.command()
 @dandiset_path_option(
     help="Top directory (local) for the dandiset, where dandi will download "
     "(or update existing) dandiset.yaml upon successful registration.  If not "

--- a/dandi/cli/cmd_upload.py
+++ b/dandi/cli/cmd_upload.py
@@ -1,16 +1,15 @@
 import click
 
-from .command import (
+from .base import (
     devel_debug_option,
     devel_option,
     instance_option,
-    main,
     map_to_click_exceptions,
 )
 from ..consts import collection_drafts
 
 
-@main.command()
+@click.command()
 # @dandiset_path_option(
 #     help="Top directory (local) of the dandiset.  Files will be uploaded with "
 #     "paths relative to that directory. If not specified, current or a parent "

--- a/dandi/cli/cmd_validate.py
+++ b/dandi/cli/cmd_validate.py
@@ -1,9 +1,9 @@
 import click
 import os
-from .command import main, map_to_click_exceptions
+from .base import map_to_click_exceptions
 
 
-@main.command()
+@click.command()
 @click.argument("paths", nargs=-1, type=click.Path(exists=True, dir_okay=True))
 @map_to_click_exceptions
 def validate(paths):

--- a/dandi/cli/command.py
+++ b/dandi/cli/command.py
@@ -2,25 +2,19 @@
 Commands definition for DANDI command line interface
 """
 
-from functools import wraps
 import logging
-import os
-from os import path as op
 
 import click
 from click_didyoumean import DYMGroup
 
-from .. import get_logger, set_logger_level
+from .base import get_logger, lgr, map_to_click_exceptions, set_logger_level
 
 from .. import __version__
-from ..consts import dandiset_metadata_file, known_instances
 
 
 # Delay imports leading to import of heavy modules such as pynwb and h5py
 # Import at the point of use
 # from ..pynwb_utils import ...
-
-lgr = get_logger()
 
 
 class LogLevel(click.ParamType):
@@ -42,127 +36,6 @@ class LogLevel(click.ParamType):
     def get_metavar(self, param):
         return "[" + "|".join(self.levels) + "]"
 
-
-# Aux common functionality
-
-
-def get_files(paths, recursive=True, recursion_limit=None):
-    """Given a list of paths, return a list of paths
-    """
-    # For now we support only individual files
-    dirs = list(filter(op.isdir, paths))
-    if dirs:
-        raise NotImplementedError(
-            "ATM supporting only listing of individual files, no recursive "
-            "operation. Was provided following directories: {}".format(", ".join(dirs))
-        )
-    return paths
-
-
-# ???: could make them always available but hidden
-#  via  hidden=True.
-def devel_option(*args, **kwargs):
-    """A helper to make command line options useful for development (only)
-
-    They will become available..."""
-
-    def wrapper(f):
-        if not os.environ.get("DANDI_DEVEL", None):
-            return f
-        else:
-            return click.option(*args, **kwargs)(f)
-
-    return wrapper
-
-
-#
-# Common options to reuse
-#
-# Functions to provide customizations where needed
-def _updated_option(*args, **kwargs):
-    args, d = args[:-1], args[-1]
-    kwargs.update(d)
-    return click.option(*args, **kwargs)
-
-
-def dandiset_id_option(**kwargs):
-    return _updated_option(
-        "--dandiset-id",
-        kwargs,
-        help=f"ID of the dandiset on DANDI archive.  Necessary to populate "
-        f"{dandiset_metadata_file}. Please use 'register' command to first "
-        f"register a new dandiset.",
-        # TODO: could add a check for correct regexp
-    )
-
-
-def dandiset_path_option(**kwargs):
-    return _updated_option(
-        "-d",
-        "--dandiset-path",
-        kwargs,
-        help="Top directory (local) of the dandiset.",
-        type=click.Path(exists=True, dir_okay=True, file_okay=False),
-    )
-
-
-def instance_option():
-    return devel_option(
-        "-i",
-        "--dandi-instance",
-        help="For development: DANDI instance to use",
-        type=click.Choice(sorted(known_instances)),
-        default="dandi",
-        show_default=True,
-    )
-
-
-def devel_debug_option():
-    return devel_option(
-        "--devel-debug",
-        help="For development: do not use pyout callbacks, do not swallow"
-        " exception, do not parallize",
-        default=False,
-        is_flag=True,
-    )
-
-
-def map_to_click_exceptions(f):
-    """Catch all exceptions and re-raise as click exceptions.
-
-    Will be active only if DANDI_DEVEL is not set and --pdb is not given
-    """
-
-    @wraps(f)
-    def wrapper(*args, **kwargs):
-        import girder_client as gcl
-        from ..girder import get_HttpError_response
-
-        try:
-            return f(*args, **kwargs)
-        # Prints global Usage: useless in majority of cases.
-        # It seems we better use it with some ctx, so it would hint in some
-        # cases to the help of a specific command
-        # except ValueError as e:
-        #     raise click.UsageError(str(e))
-        except Exception as e:
-            if isinstance(e, gcl.HttpError):
-                resp = get_HttpError_response(e)
-                if resp is None:
-                    e_str = str(e)
-                else:
-                    e_str = resp.get("message", str(e))
-            else:
-                e_str = str(e)
-            lgr.debug("Caught exception %s", e_str)
-            if not map_to_click_exceptions._do_map:
-                raise
-            raise click.ClickException(e_str)
-
-    return wrapper
-
-
-map_to_click_exceptions._do_map = not bool(os.environ.get("DANDI_DEVEL", None))
 
 #
 # Main group
@@ -226,9 +99,16 @@ def main(log_level, pdb=False):
 #
 # Commands in the main group
 #
-from .cmd_ls import ls  # noqa: E402, F401
-from .cmd_organize import organize  # noqa: E402, F401
-from .cmd_upload import upload  # noqa: E402, F401
-from .cmd_download import download  # noqa: E402, F401
-from .cmd_validate import validate  # noqa: E402, F401
-from .cmd_register import register  # noqa: E402, F401
+from .cmd_ls import ls  # noqa: E402
+from .cmd_organize import organize  # noqa: E402
+from .cmd_upload import upload  # noqa: E402
+from .cmd_download import download  # noqa: E402
+from .cmd_validate import validate  # noqa: E402
+from .cmd_register import register  # noqa: E402
+
+main.add_command(ls)
+main.add_command(organize)
+main.add_command(upload)
+main.add_command(download)
+main.add_command(validate)
+main.add_command(register)

--- a/dandi/upload.py
+++ b/dandi/upload.py
@@ -6,8 +6,7 @@ import sys
 import time
 
 
-from .cli.command import lgr
-from . import __version__
+from . import __version__, lgr
 from .utils import ensure_datetime, ensure_strtime, get_instance
 from .consts import (
     collection_drafts,

